src/ducks/classifications.js

```diff
@@ -0,0 +1,91 @@
+import apiClient from 'panoptes-client/lib/api-client.js';
+import counterpart from 'counterpart';
+import { getSessionID } from '../lib/get-session-id';
+
+//Action Types
+const SUBMIT_CLASSIFICATION = 'SUBMIT_CLASSIFICATION';
+const CREATE_CLASSIFICATION = 'CREATE_CLASSIFICATION';
+
+const initialState = {
+  classification: null,
+};
+
+const classificationReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case CREATE_CLASSIFICATION:
+      return Object.assign({}, state, {
+        classification: action.classification,
+      });
+
+    case SUBMIT_CLASSIFICATION:
+      const classification = state.classification;
+      classification.annotations.push(action.annotations);
+      classification.update({
+        completed: true,
+        'metadata.finished_at': (new Date()).toISOString(),
+        'metadata.viewport': {
+          width: innerWidth,
+          height: innerHeight
+        }
+      }).save();
+      // classification.metadata.session = getSessionID();
+
+    default:
+      return state;
+  }
+};
+
+const createClassification = () => {
+  return (dispatch, getState) => {
+    const classification = apiClient.type('classifications').create({
+      annotations: [],
+      metadata: {
+        workflow_version: getState().workflow.data.version,
```

Comment:
[major] Problem: the default value for `workflow.data` is `null`, so certain calls for `workflow.data.version` will fail. Recommended solution: check for the existence of `workflow.data` and provide a default value if that fails.